src/html.rs

```diff
@@ -245,6 +245,32 @@ impl<'a, 'b, I: Iterator<Item=Event<'a>>> Ctx<'b, I> {
     }
 }
 
+/// Iterate over an `Iterator` of `Event`s, generate HTML for each `Event`, and
+/// push it to a `String`.
+///
+/// # Examples
+///
+/// ```
+/// use pulldown_cmark::{html, Parser};
+///
+/// let markdown_str = r#"
+/// # hello
+///
+/// * alpha
+/// * beta
+/// "#;
+/// let parser = Parser::new(markdown_str);
+///
+/// let mut html_buf = String::new();
+/// html::push_html(&mut html_buf, parser);
+///
+/// assert_eq!(html_buf, r#"<p>hello</p>
```

Comment:
Ok, I debugged this a bit. The problem is that the initial `# ` in `# hello` is being stripped by the doctest mechanism. That feels wrong to me, but I'm not excited about changing the behavior, as it likely means that tests will pass depending on the rust version.

Since this is not about header syntax per se, I feel the best resolution is just to remove the `#` from the markdown source. Then I'm happy to merge, and I'll cut a 0.1.0 version (satisfying #76 in the process).